lib/Modules/HyperVCIHooks/HyperVCIHooks.psm1

```diff
@@ -0,0 +1,1538 @@
+            Initialize-GitRepository "$openstackBuild\networking-hyperv" $NETWORKING_HYPERV_GIT "master" $cherryPicks['networking-hyperv']
+        }
+    }
+
+    if ($BuildFor -eq "openstack/nova") {
+        Initialize-GitRepository "$openstackBuild\neutron" $NEUTRON_GIT $BranchName $cherryPicks['neutron']
+        Initialize-GitRepository "$openstackBuild\os-win" $OSWIN_GIT "master" $cherryPicks['os-win']
+    }
+
+    if ($BuildFor -eq "openstack/neutron") {
+        Initialize-GitRepository "$openstackBuild\nova" $NOVA_GIT $BranchName $cherryPicks['nova']
+        Initialize-GitRepository "$openstackBuild\os-win" $OSWIN_GIT "master" $cherryPicks['os-win']
+    }
+    
+    if ($BuildFor -eq "openstack/os-win") {
+        Initialize-GitRepository "$openstackBuild\nova" $NOVA_GIT $BranchName $cherryPicks['nova']
+        Initialize-GitRepository "$openstackBuild\neutron" $NEUTRON_GIT $BranchName $cherryPicks['neutron']
+    }
+    
+    if ($BuildFor -notin $standardProjects) {
+        Initialize-GitRepository "$openstackBuild\nova" $NOVA_GIT $BranchName $cherryPicks['nova']
+        Initialize-GitRepository "$openstackBuild\neutron" $NEUTRON_GIT $BranchName $cherryPicks['neutron']
+        Initialize-GitRepository "$openstackBuild\os-win" $OSWIN_GIT "master" $cherryPicks['os-win']
+    }
+}
+
+
+function Initialize-Environment {
+    Param(
+        [string]$BranchName='master',
+        [string]$BuildFor
+    )
+
+    $dirs = @($CONFIG_DIR, $BIN_DIR, $INSTANCES_DIR, $LOG_DIR, $SERVICE_DIR)
+    foreach($dir in $dirs) {
+        if (!(Test-Path $dir)) {
+            Write-JujuLog "Creating $dir folder."
+            mkdir $dir
+        }
+    }
+
+    $mkisofsPath = Join-Path $BIN_DIR "mkisofs.exe"
+    $qemuimgPath = Join-Path $BIN_DIR "qemu-img.exe"
+    if (!(Test-Path $mkisofsPath) -or !(Test-Path $qemuimgPath)) {
+        Write-JujuLog "Extracting OpenStack binaries"
+        $zipPath = Join-Path $FILES_DIR "openstack_bin.zip"
+        Expand-ZipArchive $zipPath $BIN_DIR
+    }
+  
+    $networkType = Get-JujuCharmConfig -Scope 'network-type'
+    Initialize-GitRepositories $networkType $BranchName $BuildFor
+
+    $standardProjects = 'openstack/nova', 'openstack/neutron', 'openstack/os-win'
+    if ($buildFor -in $standardProjects) {
+        $defaultProjects = $standardProjects -notmatch $buildFor
+        foreach ($project in $defaultProjects) {
+            Install-Projects $project
+        }
+        Install-NetworkType
+        Install-Projects $buildFor
+    }
+    else {
+        foreach ($project in $standardProjects) {
+            Install-Projects $project
+        }
+        if (($buildFor -eq "none") -or ($buildFor -eq "openstack/networking-hyperv")) {
+            Install-NetworkType
+        }
+        else {
+            Install-NetworkType
+            Install-Projects $buildFor
+        }
+    }
+
+    Write-JujuLog "Environment initialization done."
+}
+
+function Set-ServiceAcountCredentials {
```

Comment:
There should be a function that does exactly this in juju-powershell-modules, and it works on Nano as well.